packages/muelu/src/Smoothers/MueLu_Ifpack2Smoother_def.hpp

```diff
@@ -55,6 +55,7 @@
 #include <Tpetra_RowMatrix.hpp>
 
 #include <Ifpack2_Chebyshev.hpp>
+#include <Ifpack2_RILUK.hpp>
```

Comment:
Why does MueLu need to include these header files directly?  Shouldn't `Ifpack2::Factory` take care of this?

Reply:
I was getting a compiler error on this file after removing the header files from Ifpack2_Factory_decl.hpp, so I just put the RILUK header next to where these others already were, and that fixed the issue.  I can move these somewhere else if we need to.

Reply:
Looks like the only reason is that we want to be able to call `getNodeSmootherComplexity`?

Reply:
@jeforster Thanks for answering!  I think this is fine for now.

@cgcgcg It would be nice if Ifpack2 had a mix-in for "has a `getNodeSmootherComplexity` method."  Alternately, maybe every Ifpack2 preconditioner should implement that?  Anyway, I don't object to this PR : - ) .